src/lib/esnext.collection.d.ts

```diff
@@ -9,3 +9,49 @@ interface MapConstructor {
         keySelector: (item: T, index: number) => K,
     ): Map<K, T[]>;
 }
+
+interface ReadonlySetLike<T> {
+    /**
+     * Despite its name, returns an iterator of the values in the set-like.
+     */
+    keys(): Iterator<T>;
+    /**
+     * @returns a boolean indicating whether an element with the specified value exists in the set-like or not.
+     */
+    has(value: T): boolean;
+    /**
+     * @returns the number of (unique) elements in the set-like.
+     */
+    readonly size: number;
+}
+
+interface Set<T> {

```

Comment:
Since all of these methods are non-mutating, we might also want to add them to `ReadonlySet<T>` since `ReadonlySet<T>` serves the same purpose as what `ReadonlyArray<T>` does for `Array<T>`.

Reply:
Done.